pkgs/by-name/gp/gpg-mdp/package.nix

```diff
@@ -1,53 +1,55 @@
 {
-  fetchurl,
-  fetchpatch,
   lib,
+  fetchFromGitHub,
   stdenv,
+  nix-update-script,
+
   ncurses,
   gnupg,
 }:
 
-let
-  version = "0.7.4";
-in
-stdenv.mkDerivation {
+stdenv.mkDerivation (finalAttrs: {
   # mdp renamed to gpg-mdp because there is a mdp package already.
   pname = "gpg-mdp";
-  inherit version;
+  version = "0.7.5";
+
   meta = {
     homepage = "https://tamentis.com/projects/mdp/";
+    changelog = "https://github.com/tamentis/mdp/releases/tag/v${finalAttrs.version}";
     license = [ lib.licenses.isc ];
     description = "Manage your passwords with GnuPG and a text editor";
   };
-  src = fetchurl {
-    url = "https://tamentis.com/projects/mdp/files/mdp-${version}.tar.gz";
-    sha256 = "04mdnx4ccpxf9m2myy9nvpl9ma4jgzmv9bkrzv2b9affzss3r34g";
+
+  src = fetchFromGitHub {
+    owner = "tamentis";
+    repo = "mdp";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-Y92y70XkUbB+lhWAzEkCB/cvfUPPKIfu0yrlCS2pKn0=";
   };
-  patches = [
-    # Pull fix pending upstream inclusion for -fno-common toolchain support:
-    #   https://github.com/tamentis/mdp/pull/9
-    (fetchpatch {
-      name = "fno-common.patch";
-      url = "https://github.com/tamentis/mdp/commit/95c77de3beb96dc7c76ff36d3f3dfb18411d7c54.patch";
-      sha256 = "1j6yvjzkx31b758yav4arhlm5ig7phl8mgx4fcwj7lm2pfvzwcsz";
-    })
-  ];
+
   buildInputs = [ ncurses ];
+
+  env.NIX_CFLAGS_COMPILE = "-D_POSIX_C_SOURCE";
```

Comment:
What's the need for this? Could do with an explanatory comment, but it breaks the build on darwin so it at least needs to be made conditional.

Reply:
Hum, that was because `gpg-mdp> cmd.c:103:23: error: implicit declaration of function 'getopt'`, but it is gone, so I guess I changed something else.

Thanks for the review !

Reply:
That's exactly the error it was _causing_ on darwin :shrug: